### ISSUE TITLE
Delete the eos token in gpt dataset processor

### ISSUE THREAD

**Issue by zhaochenyang20** [neulab/prompt2model#175]
Since our decode skips the special tokens:
```
decoded_output = self.tokenizer.decode(logits, skip_special_tokens=True)
```
So the validation set and test set should not have `eos` at the end of model_output. For example, the ideal output of `Translate French to English: Bonjour ->` is `Translate French to English: Bonjour -> Hello`. So there is no need to add `eos` in the end.
